src/quantity-selector/index.ts

```diff
@@ -0,0 +1,68 @@
+import { alias, tag, Events, Selectors, Tag } from '@storefront/core';
+
+const INITIAL_INDEX = 0;
+
+@alias('selector')
+@tag('gb-quantity-selector', require('./index.html'))
+class QuantitySelector {
+
+  state: QuantitySelector.State = {
+    selected: INITIAL_INDEX,
+    options: []
+  };
+
+  init() {
+    this.getOptions(this.props.quantity);
+
+    // args for this event is a url link
+    this.flux.on(Events.URL_UPDATED, this.reset);
```

Comment:
Is reset a special keyword in the riot lifecycle? Can pass in this.setselected directly since that's what all reset does

Reply:
not a reserved word. Also I couldn't pass in setSelected bc the args for events.url_update is a url link, and I don't want to potentially handle that in setSelected.

Reply:
Oh okay so you want to make use of the default value, in this case it can be 
```
() => this.setSelected()
```